### PR TITLE
RUN-3040 moved core_state JS to TS

### PR DESCRIPTION
**Pull request by whyn07m3**
Changes:
• Typed out core state with no logic changes
• Removed unused function `remoteAppPropDecorator`
• Minor stylistic changes
• Renamed a few parameters and variables

---

Tests results:
✅  [Windows 7](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/59383deb998bc04e20cc6cd5)
✅  [Windows 8](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/59384226998bc04e20cc6cd8)
✅  [Windows 10](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/59383da9998bc04e20cc6cd3)

### CODE REVIEW

**Comment by wenjunche** [comment]
how about some comments for each class/interface.  Window, OpenFinWindow, BrowserWindow, they all look alike...

**Comment by whyn07m3** [comment]
@wenjunche Let's add those in a separate PR. I also don't have extensive knowledge to describe them, I simply typed them out, so I can't really write comments for them...

**Comment by whyn07m3** [comment]
Retested after rebasing:
✅  [Windows 7](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/5939570f998bc04e20cc6cf3)
✅  [Windows 8](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/5939573e998bc04e20cc6cf4)
✅  [Windows 10](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/593956fa998bc04e20cc6cf2)

File: src/browser/core_state.ts [hunk 1]
```diff
@@ -410,69 +412,72 @@ function getOfWindowByUuidName(uuid, name) {
...
-    return coreState.apps
+function getWinList(): Shapes.Window[] {
+    return apps
         .map(app => app.children) //with children
         .reduce((wins, myWins) => wins.concat(myWins), []); //flatten
 }
 
-function getAllApplications() {
-    return coreState.apps.map(app => {
+interface StrippedApplication {
```
Comment (HarsimranSingh) [inline src/browser/core_state.ts]: Move this to the top

File: src/browser/core_state.ts [hunk 2]
```diff
@@ -410,69 +412,72 @@ function getOfWindowByUuidName(uuid, name) {
...
-function getAllAppObjects() {
-    return coreState.apps
+export function getAllAppObjects(): Shapes.AppObj[] {
+    return apps
         .filter(app => app.appObj) //with openfin app object
         .map(app => app.appObj); //and return same
 }
 
-function getAllWindows() {
+interface StrippedWindow {
```
Comment (HarsimranSingh) [inline src/browser/core_state.ts]: Same with this one

File: src/browser/core_state.ts [hunk 3]
```diff
@@ -483,28 +488,28 @@ function shouldCloseRuntime(ignoreArray) {
...
             ignoredApps.indexOf(app.uuid) < 0 && // app is not being ignored
-            !(app._options.nonPersistent !== undefined ? app._options.nonPersistent : app._options.nonPersistant) // app is persistent
+            !app._options.nonPersistent // app is persistent
         );
     }
 }
 
 //TODO: This needs to go go away, pending socket server refactor.
-var socketServerState = {};
+let socketServerState = {};
```
Comment (HarsimranSingh) [inline src/browser/core_state.ts]: Move this up too

File: src/browser/core_state.ts [hunk 4]
```diff
@@ -13,92 +13,110 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
...
+    proxyAddress?: string;
+    proxyPort?: number;
+    type?: string;
+}
 
-// locals
-const args = electronApp.getCommandLineArguments(); // command line string ("s" for "string")
-const argv = electronApp.getCommandLineArgv(); // argument list ("v" for "vector")
-const argo = minimist(argv); // minimist-style object ("o" for "object"; hash of command line options:values; see https://github.com/substack/minimist)
+interface StrippedApplication {
```
Comment (datamadic) [inline src/browser/core_state.ts]: naming? Are these things that are found on the application, meaning is this really a subset of application?

**Comment by whyn07m3** [comment]
Retested:
✅  [Windows 7](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/594816f10c1017021736586e)
✅  [Windows 8](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/594816a80c1017021736586d)
✅  [Windows 10](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/594816f30c1017021736586f)

**Comment by HarsimranSingh** [comment]
👍

**Comment by whyn07m3** [comment]
Vanilla test results:
[Windows 7 (1 error)](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/59512bec0c101702173658a1)
[Windows 8 (3 errors)](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/59512c590c101702173658a2)
[Windows 10 (4 errors)](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/59512c7e0c101702173658a3)

This PR test results:
[Windows 7 (4 errors)](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/595126950c1017021736589e)
[Windows 8 (3 errors)](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/595126e30c1017021736589f)
[Windows 10 (4 errors)](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/5951270d0c101702173658a0)

**Pull request merged**